[SentinelOne][Security Solution] Show `observer.serial_number` info on Table view for SentinelOne alerts (instead of agent status)

ashokaditya opened the pull request:
## Summary

For SentinelOne alerts, since. 8.13, on the overview tab, we show `agent.status` field in the Highlighted Fields section. On the table view we should show the `observer.serial_number` info for that field instead of showing agent status.

**before**
![Screenshot 2024-04-17 at 1 59 53 PM](https://github.com/elastic/kibana/assets/1849116/2d196444-cbef-4308-99fd-29f34fe3ee38)

**after**
![Screenshot 2024-04-22 at 10 57 49 AM](https://github.com/elastic/kibana/assets/1849116/2b0dd093-caac-4b5d-a11e-9f8e3ce103cf)


**clip showing details on alert view and timeline views**
![sentinel-one-agent status](https://github.com/elastic/kibana/assets/1849116/75de4db9-9966-4c91-94a5-4139ea81869e)


### Checklist

- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed

<!--ONMERGE {"backportTargets":["8.14"]} ONMERGE-->

ashokaditya committed:
replace `agent.status` with `observer.serial_number`

For SentinelOne alerts, since. 8.13, on the Overview tab, we show `agent.status` field in the Highlighted Fields section.
We missed this detail on the Table tab of the flyout.
This commit fixes that.

closes elastic/security-team/issues/8896

elasticmachine commented:
Pinging @elastic/security-defend-workflows (Team:Defend Workflows)

PhilippeOberti reviewed (approved):
LGTM!

paul-tavares commented on x-pack/plugins/security_solution/public/flyout/document_details/right/tabs/table_tab.tsx:
```diff
@@ -45,7 +46,12 @@ export const getColumns: ColumnsProvider = ({
     render: (field, data) => {
       return (
-        <FieldNameCell data={data as EventFieldsData} field={field} fieldMapping={undefined} />
+        <FieldNameCell
```
Any chance you can add a Unit test for this?

ashokaditya replied:
Absolutely

ashokaditya commented:
> Thanks for addressing this @ashokaditya 👍 However this makes me think why we do have a `host health` badge in the `Table` tab. I think this should still show agent.id (observer.serial_number) and the value should be reverted from host health to string. We have other places that use this health component - eg. Overview tab and there the change would be fine.
> 
> I think that table tab should reflect the table tab.
> 
> What do you think?

Thanks for the review @tomsonpl. Yeah I agree that that would be the way to go but, we have a similar logic for endpoint alerts already. That's why I went with this solution.

See an example of Endpoint alert, as you can see, there's no `agent.status` field in the alert details.

![Screenshot 2024-04-18 at 11 27 51 AM](https://github.com/elastic/kibana/assets/1849116/163dba11-4a52-4ff2-b7e1-d544c6c65493)

I think in order to ensure that we show `agent.id`/`observer.serial_number` strings instead of agent status in the table view is a bigger discussion. I'll create a ticket for it if it is something we would like to change.

tomsonpl commented:
Thanks Ash! I am not sure if I understood, do you mean there is a similar logic (showing agents status) in endpoint alerts? I believe that the status badge in table is just a bug and tit should be fixed instead of renaming the field.

tomsonpl commented:
Take a look at this: https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/timelines/components/timeline/body/renderers/formatted_field.tsx#L276

ashokaditya commented:
> Thanks Ash! I am not sure if I understood, do you mean there is a similar logic (showing agents status) in endpoint alerts? I believe that the status badge in table is just a bug and tit should be fixed instead of renaming the field. The table/json represent the event data, not the agent status (that we have to fetch additionally), right?

I think I'm not explaining this well. I'm going to try again.

You see, the overview section for endpoint alerts also shows an `agent.status` field along with Endpoint health/isolation status. Ergo, we went with the same UX logic for sentinel one. Which is why we show an `agent.status` along with it's health/isolation status. Note that in both cases there is no `agent.status` in the alert data (as you can see in the JSON output, and as you pointed out). The code that you listed here [was recently added](https://github.com/elastic/kibana/pull/176210) to fix a [recent bug](elastic/kibana/issues/174235) that was raised to point to the fact that instead of seeing an `agent.status` we were seeing an `observer.serial_number` field in the Overview section of SentinelOne alerts.

Now, I agree that it is odd that when you search for `agent.status` in the Table view nothing shows up for either agent type alerts, but searching for `agent.id` or `observer.serial_number` shows the agent status badges. I feel this should be updated at least for the Table view search, but I think we need input from the team on this, as I am not aware why the alerts flyout shows agent status for endpoint alert. It maybe intentional and I don't want to create a regression.

@dasansol92 @paul-tavares @caitlinbetz ☝🏼

tomsonpl commented:
@ashokaditya Thanks for trying again :) 
I agree that the Overview tab should contain a badge and I already stated that before - it does already but with wrong field name, I am not sure if this PR solves this. What I am saying is that we shouldn't have this logic in the Table tab - and I pasted you a link to a line that does that. Now with your change if I am not mistaken the `observer.serial_number` field will not be searchable in the table, which should not happen while it should reflect the state of a JSON tab.

ashokaditya commented:
> @ashokaditya Thanks for trying again :) I agree that the Overview tab should contain a badge and I already stated that before - it does already but with wrong field name, I am not sure if this PR solves this. What I am saying is that we shouldn't have this logic in the Table tab - and I pasted you a link to a line that does that. Now with your change if I am not mistaken the `observer.serial_number` field will not be searchable in the table, which should not happen while it should reflect the state of a JSON tab.

You can search for `observer.serial_number` on the Table view but it will show you agent.status and status badges, same as in Overview tab. Same for endpoint alert, you can search for `agent.id` that shows the agent status/isolation badges.

What this PR solves is part of fixing [this bug](https://github.com/elastic/kibana/issues/174235), that is, if we're showing `agent.status` instead of `observer.serial_number` for SentinelOne alerts then it should be in all places in the alert, and not just one view. Yes, this does not update the JSON output which is the raw alert data, but again, like I said, this is how it is done for Endpoint alerts.

tomsonpl commented:
I am even more confused right now, this is your screenshot - it doesn't show status as id. 
<img width="1110" alt="Zrzut ekranu 2024-04-18 o 16 47 01" src="https://github.com/elastic/kibana/assets/16632552/224ff978-ad37-4b49-8390-da1f3bef1d37">

I think we need a proxy for communication here, because we're not getting anywhere :P 
@paul-tavares do you see my point by any chance?

ashokaditya commented:
> I am even more confused right now, this is your screenshot - it doesn't show status as id. 
> I think we need a proxy for communication here, because we're not getting anywhere :P @paul-tavares do you see my point by any chance?

Sorry for causing all the confusion. This one is a screenshot of Endpoint alert on the Table view for context. When you search for `agent.id` (in the screenshot I searched for `agent` to show all matching fields in alert data), you can clearly see that there is no `agent.status`. Hope that clears up some confusion. 😅

ashokaditya committed:
Revert "replace `agent.status` with `observer.serial_number`"

This reverts commit d48d1a39105d5d56d532741d48cdadf60b678486.

ashokaditya committed:
Merge branch 'main' into fix/dw-sentinel-one-alert-table-details-bug-8896

ashokaditya committed:
Merge branch 'main' into fix/dw-sentinel-one-alert-table-details-bug-8896

ashokaditya committed:
show `observer.serial_number` on table view

For sentinel one alerts, instead of showing agent status on table view for `observerl.serial_number` show the serial_number info.

closes elastic/security-team/issues/8896

ashokaditya commented:
@tomsonpl we talked offline and agreed to show the `observer.serial_number` info instead of agent status. Thanks for catching this in the previous commit. The new commit here now implements that.

tomsonpl reviewed (approved):
lgtm, thanks!

tomsonpl commented:
Could you by the way double check what fieldName:value is rendered in the Overview tab? Thanks!

ashokaditya committed:
Show agent status info on timeline overiew

ashokaditya committed:
Merge branch 'main' into fix/dw-sentinel-one-alert-table-details-bug-8896

ashokaditya commented on x-pack/plugins/security_solution/public/timelines/components/timeline/body/renderers/formatted_field.tsx:
```diff
@@ -270,7 +270,7 @@ const FormattedFieldValueComponent: React.FC<{
     );
   } else if (
-    fieldName === SENTINEL_ONE_AGENT_ID_FIELD ||
+    fieldName === AGENT_STATUS_FIELD_NAME &&
```
This is the change that ensures that we show the agent status agains agent.status field and not `observer.serial_number` field.

tomsonpl replied:
Interesting! Could you explain what is the difference between fieldName and fieldFromBrowserField?.name ? I am wondering if we will ever meet the following condition. 
Thanks 🙇

ashokaditya replied:
As I understand it on the surface `fieldName` is what tells the component which field name to show on the table. This may be the actual field name or may have been overridden by some other custom string. `fieldFromBrowserField` is what the actual field name. 

Not sure what you mean by ever meeting the condition? 

With some testing it was clear that the current condition is too loose and thus shows agent status component for conditions when `fieldName: observer.serial_number` and `fieldFromBrowserField.name: observer.serial_number` (in table tab view) when in reality it should show the sentinel agent status component when `fieldName: agent.status` and `fieldFromBrowserField.name: observer.serial_number` (when on overview tabs).

tomsonpl replied:
Thank you for explanation 👍 
I hope you don't mind if I dig a bit deeper here :P 
Does this change actually applies to `Overview tab`?  This again will show `agent.status` healthy badge in `Table tab` which I think won't ever happen since normally we do not have agent.status field in events, or am I wrong?

tomsonpl replied:
When we're looking at `observer.serial_number` it shows correctly a string 👍 
But If I am not wrong, the code change above would show a healthy badge in `Table tab` next to `agent.status` field, which I believe does not exist, thus this change is redundant. 
Were you able to reproduce a situation where we have `fieldName: agent.status` and `fieldFromBrowserField.name: observer.serial_number` ?

tomsonpl replied:
We discussed this offline, the `fieldName: agent.status` and `fieldFromBrowserField.name: observer.serial_number` is met on the `Timelines Overview tab` - however not happening in Alert Overview tab. 

So the change makes sense. Thanks for doing this @ashokaditya :)

tomsonpl reviewed (approved):
LGTM 👍 Nice catch with the timeline overview tab 👏

ashokaditya committed:
Merge branch 'main' into fix/dw-sentinel-one-alert-table-details-bug-8896
